core/src/test/java/org/elasticsearch/search/SearchModuleTests.java

```diff
@@ -99,16 +99,16 @@ public void testRegisteredQueries() throws IOException {
         IndicesQueriesRegistry indicesQueriesRegistry = module.getQueryParserRegistry();
         XContentParser dummyParser = XContentHelper.createParser(new BytesArray("{}"));
         for (String queryName : supportedQueries) {
-            indicesQueriesRegistry.lookup(queryName, dummyParser, ParseFieldMatcher.EMPTY);
+            indicesQueriesRegistry.lookup(queryName, ParseFieldMatcher.EMPTY, dummyParser::getTokenLocation);
```

Comment:
working on that. I just noticed that we end up with a few more object not using the Supplier. Because the jackson parsers create a new location object on demand, and we create our own based on that. Still convinced we should go that route?